libbindgen/src/ir/layout.rs

```diff
@@ -31,4 +35,32 @@ impl Layout {
     pub fn zero() -> Self {
         Self::new(0, 0)
     }
+
+    /// Get this layout as an opaque type.
+    pub fn opaque(&self) -> Opaque {
+        Opaque(*self)
+    }
+}
+
+/// When we are treating a type as opaque, it is just a blob with a `Layout`.
+pub struct Opaque(pub Layout);
+
+impl CanDeriveDebug for Opaque {
+    type Extra = ();
+
+    fn can_derive_debug(&self, _: &BindgenContext, _: ()) -> bool {
+        self.0.size < RUST_DERIVE_IN_ARRAY_LIMIT
```

Comment:
This is not right (or at least, not as permissive as it should). Size is in bytes, but we sometimes generate arrays of `u32`, `u64` to preserve alignment.

This code should check that to decide it too.